src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,179 @@
+     */
+    @SuppressWarnings("unchecked")
+    public void saveData(TaskList taskList) {
+        File savedData = new File(filePath);
+
+        //Add employees to list
+        JSONArray jsonTaskList = new JSONArray();
+
+        for (int i = 0; i < taskList.getTotalNumberOfTask(); i++) {
+            Task currTask = taskList.getTaskById(i);
+            JSONObject taskDetails = new JSONObject();
+            String type = currTask.getType().toString();
+            taskDetails.put("type", type);
+            taskDetails.put("value", currTask.getValue());
+            taskDetails.put("time", currTask.getTime() == null ? "" : currTask.getTime().toString());
+            taskDetails.put("done", currTask.isDone());
+
+            JSONObject taskObj = new JSONObject();
+            taskObj.put("task", taskDetails);
+            jsonTaskList.add(taskObj);
+        }
+
+        try {
+            savedData.delete();
+            savedData.createNewFile();
+        } catch (Exception e) {
+            System.out.println("An error occurred");
+        }
+
+        //Write JSON file
+        try (FileWriter file = new FileWriter(filePath)) {
+            //We can write any JSONArray or JSONObject instance to the file
+            file.write(jsonTaskList.toJSONString());
+            file.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Helper function that conver json to List of Task objects.
+     *
+     * @param fileName The name of the file.
+     * @return The list of Tasks.
+     */
+    @SuppressWarnings("unchecked")
+    public List<Task> convertJsonToTaskListObj(String fileName) {
+        List<Task> taskList = new ArrayList<>();
+
+        //JSON parser object to parse read file
+        JSONParser jsonParser = new JSONParser();
+        String dirPath = System.getProperty("user.dir") + "\\src\\main\\data";
+        try (FileReader reader = new FileReader(dirPath + "\\" + fileName)) {
+            //Read JSON file
+            Object obj = jsonParser.parse(reader);
+
+            JSONArray taskArr = (JSONArray) obj;
+
+            //Iterate over task array
+            taskArr.forEach(task -> taskList.add(parseTaskObject((JSONObject) task)));
+
+        } catch (IOException | ParseException e) {
+            e.printStackTrace();
+        }
+
+        return taskList;
+
+    }
+
+    /**
+     * Convert JSONObject into Task.
+     *
+     * @param task JSONObject that contains information about task.
+     * @return A new Task object.
+     */
+    private Task parseTaskObject(JSONObject task) {
+        //Get task object within list
+        JSONObject taskObject = (JSONObject) task.get("task");
+
+        String value = (String) taskObject.get("value");
+        boolean done = (boolean) taskObject.get("done");
+        String time = (String) taskObject.get("time");
+        String type = (String) taskObject.get("type");
+        type = type.toUpperCase();
```

Comment:
I like that you abstracted out the conversion of the type to UpperCase which allowed for cleaner looking code in lines 165-171! :D